packages/mermaid/src/dagre-wrapper/edges.js

```diff
@@ -435,22 +448,59 @@ export const insertEdge = function (elem, e, edge, clusterDb, diagramType, graph
   const lineData = points.filter((p) => !Number.isNaN(p.y));
 
   // This is the accessor function we talked about above
-  let curve;
+  let curve = curveBasis;
   // Currently only flowcharts get the curve from the settings, perhaps this should
   // be expanded to a common setting? Restricting it for now in order not to cause side-effects that
   // have not been thought through
-  if (diagramType === 'graph' || diagramType === 'flowchart') {
-    curve = edge.curve || curveBasis;
-  } else {
-    curve = curveBasis;
+  if (edge.curve && (diagramType === 'graph' || diagramType === 'flowchart')) {
+    curve = edge.curve;
   }
-  // curve = curveLinear;
+
+  // The offsets are calculated from the markers' dimensions.
```

Comment:
```suggestion
  // We need to draw the lines a bit shorter to avoid drawing
  // under any transparent markers.
  // The offsets are calculated from the markers' dimensions.
```